internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -904,6 +995,24 @@ func checkNewSchemaCompatibility(specs []any) bool {
 	return true
 }
 
+// When legacy schema structure is used we invoke the old API for updates. This API sends diskSizeGB at root level.
+// This function is used to detect if changes are made in the inner spec levels. It assumes that all disk_size_gb values at the inner spec level have the same value, so it looks into first region config.
```

Comment:
What happens if the users try to set different disk_size_gb values? Does the API handle that?

Reply:
The new API will handle validation in case disk_size_gb defines different values (https://jira.mongodb.org/browse/CLOUDP-263950). 
Only case where users would not have a clear message is if using old structure (numShards > 1), and place different disk size gb inner values. In this case we would take the inner value from first replication spec for the update, so a following plan would show that other disk_size_gb values was not updated.
Added the mention `This value must be equal for all shards and node types.` in the docs as part of this PR to avoid any confusion.